src/Model/List/ArrayList.php

```diff
@@ -836,13 +809,13 @@ protected function normaliseFilterArgs($column, $value = null)
      *
      * @return static<T>
      */
-    public function byIDs($ids)
+    public function byIDs(array $ids): static
     {
         $ids = array_map('intval', $ids ?? []); // sanitize
         return $this->filter('ID', $ids);
     }
 
-    public function byID($id)
+    public function byID(int|string|null $id): mixed
```

Comment:
Unlike with `DataList`, ID could be anything, since any object/assoc array can be stored in an `ArrayList`.
```suggestion
    public function byID(mixed $id): mixed
```

Reply:
OK, though this is for the $id, why would you use something that's not an int or a string as an $id?

Reply:
This is for a list which can contain any values for any reason. It should be `mixed` here because "ID" isn't something we control here. It could be an object of a `Uuid` class, for example.

Reply:
Changed to mixed